docs/backend.md

```diff
@@ -0,0 +1,138 @@
+
+## API
+
+The data model and storage are likely the difficult parts of this design; the API and server implementing it should be much simpler. They can just be a thin **stateless** wrapper, performing authentication and then interacting with the database. Thus, they could be serverless, since the start-up time should be trivial.
+
+We should check if GDS prefers any particular languages or frameworks. Some options that some/all of the team is familiar with:
+
+- Go
+- Python (Flask?) -- is the GIL a problem?
+- Rust (overkill, Dustin not in favor)
+- Others?
+
+We can figure out the JSON (or gRPC?) API as we go. The output would generally be an ATIP GeoJSON file, or the currently-evolving combined scheme GeoJSON format. Some endpoints would be:
+
+- get one scheme + version
+- get the latest version of a scheme
+- update one scheme
+- get all schemes combined (possibly filtering the results)
+
+### Svelte / UI changes
+
+We need to integrate whatever GDS auth layer. Talking to the backend may be trivial (`fetch` calls), or maybe we use something like websockets for saving commands while people edit.
+
+We should be very careful to keep the existing ATIP pages (browsing authorities, editing a scheme, and browsing a bunch of combined schemes) able to be used **without** any of the backend and database this doc describes. The current file-based approach is very simple to deploy. There may be non-ATE users who don't want any sort of backend. So, we need to find a way to make the frontend changes agnostic to a backend (the current one being "local storage").
+
+Similarly, whatever backend we develop will inevitably be coupled to ATE use cases, scheme metadata, etc. Making it generic to allow another org to someday deploy it is not high priority. All the same, we should future-proof this from the start by keeping all the backend code open source from the start, using Docker and Terraform and similar to provision cloud resources based on config that somebody could easily adjust. There will be exceptions (like GDS auth integration), but we should try to keep things reasonably modular.
+
+### Syncing
+
+Suppose somebody is editing an ATIP scheme. How often should the frontend save the changes to the cloud? Should this only happen when the user explicitly saves and submits a version of the scheme? Should we log every command/edit and stream those to a database with something like websockets?
```

Comment:
I need to think about this one. 

It feels like only updating the database would be when the user explicitly saves/submits but it might be nice to log commands/edits so that progress is saved without submitting.

Reply:
Even if it's something simple like the same person switching laptops and expecting their intermediate work to be visible, auto-saving everything could be useful and/or expected by users